keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/DoubleTensor.java

```diff
@@ -177,6 +177,8 @@ static DoubleTensor placeHolder(int[] shape) {
 
     double determinant();
 
+    double product();
```

Comment:
Is this for the Dirichlet?

Reply:
Didn't need it in the end but thought it would be worth keeping